x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/BlobCacheUtilsTests.java

```diff
@@ -25,6 +27,56 @@ public void testReadSafeThrows() {
     public void testToPageAlignedSize() {
         long value = randomLongBetween(0, Long.MAX_VALUE - SharedBytes.PAGE_SIZE);
         long expected = ((value - 1) / SharedBytes.PAGE_SIZE + 1) * SharedBytes.PAGE_SIZE;
-        assertThat(BlobCacheUtils.toPageAlignedSize(value), Matchers.equalTo(expected));
+        assertThat(BlobCacheUtils.toPageAlignedSize(value), equalTo(expected));
+        assertThat(BlobCacheUtils.toPageAlignedSize(value), equalTo(roundUpUsingRemainder(value, SharedBytes.PAGE_SIZE)));
+    }
+
+    public void testRoundUpToAlignment() {
+        assertThat(BlobCacheUtils.roundUpToAlignedSize(8, 4), equalTo(8L));
+        assertThat(BlobCacheUtils.roundUpToAlignedSize(9, 4), equalTo(12L));
+        long alignment = randomLongBetween(1, Long.MAX_VALUE / 2);
+        long value = randomLongBetween(0, Long.MAX_VALUE - alignment);
```

Comment:
I think a value of `0` does _not_ work except when `alignment = 1`?

Reply:
Correct, I chose to special case 0 to handle it in 32c3795d7f26906c6f47dd9d1fa621751010544a, just to avoid surprises.